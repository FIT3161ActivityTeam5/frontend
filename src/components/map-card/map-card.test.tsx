import React from 'react';
import renderer from 'react-test-renderer';

import MapCard from './map-card';

describe('<MapCard />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MapCard mapId="test_id" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
