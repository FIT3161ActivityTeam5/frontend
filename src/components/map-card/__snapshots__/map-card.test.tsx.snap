// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapCard /> renders correctly 1`] = `
Array [
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(249, 250, 251, 1)",
          "borderBottomLeftRadius": 6,
          "borderBottomRightRadius": 6,
          "borderTopLeftRadius": 6,
          "borderTopRightRadius": 6,
          "flexDirection": "row",
          "marginBottom": 8,
          "marginLeft": 16,
          "marginRight": 16,
          "marginTop": 8,
          "overflow": "hidden",
        },
        Object {
          "elevation": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.18,
          "shadowRadius": 1,
        },
      ]
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbWidth="33.333333%"
      focusable={false}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Array [
            Object {
              "borderBottomColor": "rgba(209, 213, 219, 1)",
              "borderLeftColor": "rgba(209, 213, 219, 1)",
              "borderRightColor": "rgba(209, 213, 219, 1)",
              "borderRightWidth": 1,
              "borderTopColor": "rgba(209, 213, 219, 1)",
              "width": "33.333333%",
            },
            Object {
              "aspectRatio": 1,
            },
          ],
        ]
      }
      vbHeight={1024}
      vbWidth={1024}
    >
      <RNSVGGroup>
        <RNSVGCircle
          cx={512}
          cy={512}
          propList={
            Array [
              "stroke",
              "strokeWidth",
            ]
          }
          r={128}
          stroke={4291941851}
          strokeWidth={9}
        />
        <RNSVGCircle
          cx={512}
          cy={512}
          propList={
            Array [
              "stroke",
              "strokeWidth",
            ]
          }
          r={384}
          stroke={4291941851}
          strokeWidth={9}
        />
        <RNSVGLine
          propList={
            Array [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={4291941851}
          strokeWidth={9}
          x1={512}
          x2={512}
          y1={0}
          y2={1024}
        />
        <RNSVGLine
          propList={
            Array [
              "stroke",
              "strokeWidth",
            ]
          }
          stroke={4291941851}
          strokeWidth={9}
          x1={0}
          x2={1024}
          y1={512}
          y2={512}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <View
      style={
        Object {
          "flexDirection": "column",
          "flexGrow": 1,
          "justifyContent": "space-between",
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins_700Bold",
              },
              Object {
                "fontSize": 20,
                "lineHeight": 28,
              },
            ]
          }
        >
          Test Map
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins_400Regular",
              },
              Object {},
            ]
          }
        >
          Map with 
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins_400Regular",
                },
                Object {},
              ]
            }
          >
            0
          </Text>
           nodes and 
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins_400Regular",
                },
                Object {},
              ]
            }
          >
            0
          </Text>
           edges.
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(124, 58, 237, 1)",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "justifyContent": "center",
              "marginRight": 4,
              "opacity": 1,
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins_700Bold",
                },
                Object {
                  "color": "rgba(255, 255, 255, 1)",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            Delete
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(124, 58, 237, 1)",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 8,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 8,
              "width": 40,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins_700Bold",
                },
                Object {
                  "color": "rgba(255, 255, 255, 1)",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            ...
          </Text>
        </View>
      </View>
    </View>
  </View>,
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(17, 24, 39, 0.75)",
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingBottom": 32,
          "paddingLeft": 32,
          "paddingRight": 32,
          "paddingTop": 32,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "rgba(249, 250, 251, 1)",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins_600SemiBold",
              },
              Object {
                "color": "rgba(124, 58, 237, 1)",
                "fontSize": 24,
                "lineHeight": 32,
              },
            ]
          }
        >
          Edit Map
        </Text>
        <TextInput
          allowFontScaling={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Enter some text here"
          placeholderTextColor="white"
          rejectResponderTermination={true}
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 1)",
              "borderBottomWidth": 2,
              "borderLeftWidth": 2,
              "borderRightWidth": 2,
              "borderTopWidth": 2,
              "color": "rgba(55, 65, 81, 1)",
              "height": 32,
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "paddingLeft": 8,
              "paddingRight": 8,
              "textAlign": "justify",
              "width": "75%",
            }
          }
          underlineColorAndroid="transparent"
          value="Test Map"
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(124, 58, 237, 1)",
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 12,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 12,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "Poppins_700Bold",
                },
                Object {
                  "color": "rgba(255, 255, 255, 1)",
                  "fontSize": 16,
                  "lineHeight": 24,
                },
              ]
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </Modal>,
]
`;
